src/js/components/Diagram/index.d.ts

```diff
@@ -5,14 +5,15 @@ import {
 } from "../../utils";
 export interface DiagramProps {
   connections: {
-    anchor?: "center" | "vertical" | "horizontal", 
+    anchor?: "center" | "vertical" | "horizontal" | string, 
```

Comment:
This will fix the issue I see in the editor/storybook! The only thing is that now it's not taking advantage of a type system since any string is valid but not necessarily meaningful

that is, "center" is now the same as "asfasdfsfd" (which is, i think actually, the same as `undefined` as well).
https://github.com/grommet/grommet/blob/master/src/js/components/Diagram/Diagram.js#L152

if there were exported type aliases defined for these unions, it would allow easily allow the cosumer to define the type on their own object and enforce a valid value.

```ts
export type DiagramConnectionAnchor = "center" | "vertical" | "horizontal"
export type DiagramConnectionType = "direct" | "curved" | "rectilinear"
```
it would also be nice to have the connection object available easily since a `Connection` type is really the 🍗 and 🥔 s here.

🤷‍♀ I dunno if you folks are building out the type system further or no?

Reply:
AGREED!